infrastructure/logger/utils.go

```diff
@@ -15,3 +16,11 @@ func LogAndMeasureExecutionTime(log *Logger, functionName string) (onEnd func())
 		log.Debugf("%s end. Took: %s", functionName, time.Since(start))
 	}
 }
+
+// LogMemoryStats logs memory stats for `functionName`
+func LogMemoryStats(log *Logger, functionName string) {
+	stats := runtime.MemStats{}
+	runtime.ReadMemStats(&stats)
```

Comment:
I think we want to do this in a logClosure.
Reading the code of ReadMemStats it starts with the line
`stopTheWorld("read mem stats")`

So I think it's better not to run it if not necassary.